### PR TITLE
chore: update decorators from backoff python module

### PR DESCRIPTION
**Pull request by jagadeesh545**
## Description

`backoff.on_exception()` method in the logging script is failing due to invalid arguments. Please find the stack trace below.
```
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/run/determined/enrich_task_logs.py", line 121, in run
    self.ship()
  File "/opt/conda/lib/python3.7/site-packages/backoff/_sync.py", line 92, in retry
    wait = _init_wait_gen(wait_gen, wait_gen_kwargs)
  File "/opt/conda/lib/python3.7/site-packages/backoff/_common.py", line 30, in _init_wait_gen
    initialized.send(None)  # Initialize with an empty send
AttributeError: 'float' object has no attribute 'send'
```
After investigation, it was identified that the reason for this failure is that the first argument to the `backoff.on_exception()` method was a `lambda` function instead of a `generator` (shown below).
```
@backoff.on_exception(  # type: ignore
        lambda: backoff.full_jitter(SHIPPER_FAILURE_BACKOFF_SECONDS),
        errors.APIException,
        max_tries=3,
    )
```
Examples of correct usage of the `backoff.on_exception()` method can be found [here](https://github.com/litl/backoff/tree/master).
The first argument should be one of the generators - expo, fibo, constant, runtime, as shown in the documentation mentioned above.
An example of the correct usage is provided below.
```
@backoff.on_exception(  # type: ignore
        backoff.constant,
        errors.APIException,
        max_tries=3,
    )
```
This PR provides corrections for the `backoff.on_exception()` method.

## Test Plan

Tested manually.

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by stoksc — APPROVED** [review]
lgtm

**Comment by stoksc** [comment]
This library is unpinned in envs and v1 -> v2 had a breaking change

**Comment by stoksc** [comment]
I think we can close this. Iirc it was already fixed somehow or another.

**Comment by jagadeesh545** [comment]
This is still an open issue. I added a test in the file `master/static/srv/test_enrich_task_logs.py`

File: master/static/srv/check_ready_logs.py [hunk 1]
```diff
@@ -16,13 +16,22 @@
...
+
 # Since the service is virtually inaccessible by the user unless
 # the call completes, we may as well try forever or just wait for
 # them to kill us.
 @backoff.on_exception(  # type: ignore
-    lambda: backoff.constant(interval=BACKOFF_SECONDS),
+    backoff.constant,
     RequestException,
     giveup=lambda e: e.response is not None and e.response.status_code < 500,
+    interval=BACKOFF_SECONDS,
```
Comment (stoksc) [inline master/static/srv/check_ready_logs.py:34]: i think we should also set max_tries.

Comment (stoksc) [reply master/static/srv/check_ready_logs.py:34]: and we need to pin the version in the setup.py file still so this doesn't happen again if they make breaking changes

Comment (jagadeesh545) [reply master/static/srv/check_ready_logs.py:34]: Added max_tries and added requirements file to pin the `backoff` package version to `1.10.0`. Additionally, set up the CircleCI job `test-unit-master-python` inside the `test-unit` workflow to test the new python test.

**Comment by jagadeesh545** [comment]
PR is ready to merge. Please merge it.

File: master/static/srv/check_ready_logs.py (hunk 1 above)
Comment (stoksc) [reply master/static/srv/check_ready_logs.py:34]: when i install i get >2, why pin to 1.10.0?

Comment (jagadeesh545) [reply master/static/srv/check_ready_logs.py:34]: I had 1.10.0 installed on my system. I didn't realize there was version 2.x. I will test with 2.x and update the PR.

Comment (jagadeesh545) [reply master/static/srv/check_ready_logs.py:34]: pinned backoff to the latest version `2.2.1`

**Comment by rb-determined-ai** [comment]
I'll be taking over the ml-sys review of this one for @azhou-determined.

@jagadeesh545, @stoksc what's the current status on this one?  `enrich_logs.py` is gone now, for instance.

**Comment by azhou-determined** [comment]
the `backoff` module was removed in https://github.com/determined-ai/determined/pull/6590, since it was only doing constant backoff. i think this PR can be closed.

**Comment by rb-determined-ai** [comment]
cool, also I heard from @jagadeesh545 out-of-band, he agrees it can be closed.

**Pull request closed without merging**